doc/configuration.rst
Configuration
=============

This part of the documentation will deal with the configuration options in the installation routine.

.. note::
    
    Please note that some terms which are used here may be translated into another language which you selected before.

Mainsettings-Tab
----------------

**Serveradress**
    Description
        This is the adress of your TeamSpeak3 Server
    Values
        IP-Adress or hostname (e.g. 127.0.0.1)

**Queryport**
    Description
        The port of the TeamSpeak3 Query of your server
    Values
        10011 by default

**Serverport**
    Description
        The port of the TeamSpeak3 Server which you want to monitor
    Values
        9987 by default

**Login required**
    Description
        If a login with a Query user is required to obtain the needed rights of the TeamSpeak3 Webviewer
    Values
        Yes by default

**Username**
    Username of the Query-User

**Password**
    Password of the Query-User. Please note that the password will be saved in **plain text**. We strongly recommend creating a **seperate account** for the TeamSpeak3 Webviewer.

Modules-Tab
-----------

In the modules tab you can find two columns. In the left column are the modules displayed which are enabled, in the right column the disabled ones.
Simply drag and drop the modules. To edit the configuration of a module, click on a modules name.

Style-Tab
---------

**Download Servericons automatically**
    If custom servericons should be downloaded automatically.

**Imagepack for icons**
    The imagepack which should be used for icons if you disabled the option above.

**Stylesheet**
    The stylesheet which should be used for the webviewer.

**Display arrows**
    If arrows should be displayed next to channels like in the TeamSpeak3 Client

**Show Images**
    If all icons on the right side, like moderated icon, group icon, ... should be displayed.

**Show Country Icons**
    If country icons should be shown as in the TeamSpeak3 Client

Caching-Tab
-----------

**Enable Caching**
    If the TeamSpeak3 Webviewer should cache the data it receives. We strongly recomment enabling caching for stability reasons.

**Caching Method**
    Description
        The method which should be used to cache the data. Some options may not be available on your webserver
    Values
        - APC-Cache
        - File-Cache

**Standard cachetime**
    How long received data should be stored (in seconds). We recommend 60.

Misc-Tab
--------

**Language**
    The language of the TeamSpeak3 Webviewer

**Date/Time Format**
    The format in which dates and times should be outputted. For detailed information take a look for the `PHP date() function`_.

**Provide Usage Statistics**
    This will send the URL of your TeamSpeak3 Webviewer Installation to devMX.

.. _PHP Date() function: http://php.net/manual/en/function.date.php

changelog.txt
###############
## Changelog ##
###############

! Major Change
+ Add
- Removal
~ Minor Change
* Fix

Translations:
+ German (by devMX)
+ Englisch (by devMX)
+ Portuguese (Brazil) (by Flavio)
+ French (by blablabla)

v1.4.2
- Issue #94: Removed outdated shell scripts
- Issue #97: Removed global configuration from installscript (Deprecated)
* Issue #101: Fixed l10n in installscript
* Issue #100: Fixed scollbars in loginButton and jsLogin

v1.4.1
- Issue #91: Support of default configfile. Configfiles called 'default.xml' won't be loaded automatically if present.
~ Issue #93: Added flags to language selections
~ Issue #90: Set cursor to pointer on collapsable channels and spacers, if channelHiding module is enabled
~ Updated design of installscript, welcome screen and added new jQueryUI theme
* Issue #89: Fixed welcomescreen scoll bug if too much configfiles were created
* Issue #88: Added date formatting to the serverInfo module.
* Issue #95: Fixed html bug if "Channelclientsonly" was enabled (Thanks to Tazzernator via Github.com)

v1.4
+ Issue #69: Added option to hide all icons on the right side
+ Issue #68: Added option to display country icons
+ Issue #83: Added support for APC-Cache
! Issue #74: Removed old caching and added a new method which caches the whole HTML-Output
    + Issue #83: Added APC-Cache
~ Issue #72: Added div element at ajax code output in welcome screen
~ Issue #73: Moved country flag into title bar in the infoDialog
~ Issue #76: Added effect to infoDialog, sorted and added new options
~ Issue #77: Added a hover delay for the infoDialog
~ Issue #71: Fixed PHP 5.2 compatibility with ajax features
~ Issue #80: Added a option for the formatting of dates and times
~ Issue #82: Added browser caching for images, css and js
~ Issue #81: Added a option to select the language in the code creation dialog
* Issue #75: Fixed wrong position on infoDialog
* Issue #79: L10n was not working in the module config-dialog

v1.3.2
~ Issue #57: Clienticons (set via permission system) are now displayed, too
* Issue #56: Updated all styles that they don't apply styles at global elements like <body> or <p>
* Issue #55: transfered all hardcoded module-css styles into the css file of the modules
* Issue #48: Spam of warning messages in apache log (thanks to Samifighter)
* Issue #61: Server- and Channelgrouicons are now displayed in the right order
* Issue #59: All client states like talking and channelcommander are displayed correctly now
* Issue #60: Fixed bug that some script and style files were included twice
* Issue #58: Fixed display issue with custom spacers like [*spacer1]♥. The chars were not repeated properly.
* Issue #65: Fixed bug in internet explorer where only cryptical chars where shown in the infoDialog
* Issue #66: Fixed bug of failing language selection on gettext enabled webserver
* Issue #67: Using of options in the function json_encode raised required php version to 5.3
* Issue #62: Fixed bug in channelhiding where channels faded out and in again immediatly

v1.3.1
~ Issue #54: Viewer uses no jQuery and not $ for jQuery to prevent conflicts with other libs
* Issue #52: deleted chmod.sh and added secure-permissions.sh and setup-permissions.sh ( Thanks to brrrt (forum.teamspeak.com))
* Issue #53: Bugs in modules htmlframe and autorefresh, which caused a notice

v1.3
+ new iconpackage (devMX Shield-Icons)
! Issue #49: The viewer can be included via ajax too (for more informatio see docs)
* Issue #47: Fixed div-code failure (thanks to outlaws-clan.eu)
* Updated jQuery to 1.7.1 and jQueryUI to 1.8.16
* Issue #50: Missing " in 'get code to include' (Thanks to Marco (konzil-gilde.de))
* Issue #51: js_Login didn't show the link to ts if user-choosed username was enabled (Thanks to Marco (konzil-gilde.de))

v1.2
! Issue #46: Added some options for filtering the viewer, for example show only clients or only channels with clients inside
~ Issue #45: Added a dialog which shows the code to include the viewer into a webpage
* Issue #43: TS Query Groups and wrong server groups were displayed as a users group in the infoDialog module
* Issue #44: The client version in the infoDialog was shown as an unix timestamp. The build has been removed now.

v1.1
+ Issue #25: added a cron file to update the statistics
~ Issue #33: added button to disable or enable all modules in the installscript
~ Issue #30: redesigned loginButton and js_login
~ Issue #17: redesigned welcome screen and installscript
~ Issue #31: Added opt-in usage statistics. They can be enabled in the installation script
* Issue #32: Flushing cache in installscript did not work
* Issue #34: If the password is preset in the configfile the js_login and loginButton module are crashing
* Issue #20: added custom jQueryUI Theme which will be uses as default
* Issue #37: arrows are shown wrong if channel is a spacer
* Issue #39: The l10n didn't work properly or at all if the php gettext module was enabled (Thanks to xiloid (http://forum.teamspeak.com))
* Issue #38: A whitescreen in the installscrip occured when the php DOM extension was not enabled (Thanks to xiloid (http://forum.teamspeak.com))
* Issue #36: If a login to the query is required the icon download didn't login itself
* Issue #41: The infoDialog shows now the name of the server- and channelgroups near to the icon
* Issue #35: added a bash script to automatically set chmod
* Issue #40: closing html and body tag was missing
* Issue #42: added links to the modules documentation in the installscript

v 1.0
! There are two module configfiles now:
    - one global one for all viewer-configfiles
    - one for each viewer-configfile. All options of this configfile will override the global ones
+ style 'modern-dark'
+ permission check in installscript
+ Issue #2: Display of servericon and default channel icon

install/html/config.php
<?php
/**
 *  This file is part of devMX TeamSpeak3 Webviewer.
 *  Copyright (C) 2011 - 2012 Max Rath and Maximilian Narr
 *
 *  devMX TeamSpeak3 Webviewer is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  TeamSpeak3 Webviewer is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with devMX TeamSpeak3 Webviewer.  If not, see <http://www.gnu.org/licenses/>.
 */
?>

<!-- Navigation -->
<span style="position: fixed; right: 10px; top: 10px;" class="topnav" >
    <a href="index.php?action=return" alt="" class="button"><?php __e('Back to configfiles') ?></a>
    <a href="index.php?action=logout" alt="" class="button"><?php __e('Logout') ?></a>
    <a href="http://docs.devmx.de/teamspeak3-webviewer/" title="<?php __e('I need help!') ?>" target="_blank" alt="" class="button"><span class="ui-icon ui-icon-info">&nbsp;</span></a>
</span>
<div class="tswv-container tswv-container-60per">
    <form action="index.php?action=submit" method="post" >
        <div id="tabs">
            <ul>
                <li><a href="#tab1"><?php __e('Mainsettings') ?></a></li>
                <li><a href="#tab2"><?php __e('Modules') ?></a></li>
                <li><a href="#tab3"><?php __e('Style') ?></a></li>
                <li><a href="#tab4"><?php __e('Caching') ?></a></li>
                <li><a href="#tab5"><?php __e('Misc') ?></a></li>
            </ul>

            <!-- Serverinformation -->
            <div id="tab1">
                <table class="config" cellspacing="0">
                    <tr>
                        <td><?php __e('Serveradress') ?></td>
                        <td><input type="text" value="<?php echo($data['serveradress_value']) ?>" name="serveradress" /></td>
                        <td class="option-descr"><?php __e('The adress with which you are connecting to the server (Can be an IP-Adress or hostname).') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Queryport') ?></td>
                        <td><input type="text" value="<?php echo($data['queryport_value']) ?>" name="queryport" /></td>
                        <td class="option-descr"><?php __e('The Queryport of your server (Default: 10011).') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Serverport') ?></td>
                        <td><input type="text" value="<?php echo($data['serverport_value']) ?>" name="serverport" /></td>
                        <td class="option-descr"><?php __e('The port through which you are connecting to your server via the client (Default: 9987).') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Login required') ?></td>
                        <td>
                            <input id="login-needed-true" type="radio" name="login_needed" value="true" <?php if ($data['login_needed'] != "false"): ?>checked="checked"<?php endif; ?>> <?php __e('Yes') ?><br>
                            <input id="login-needed-false" type="radio" name="login_needed" value="false" <?php if ($data['login_needed'] == "false"): ?>checked="checked"<?php endif; ?>><?php __e('No') ?>
                        </td>
                        <td class="option-descr"><p><?php __e('If a login is required that the viewer can get the needed information (Default: yes).') ?></p>
                            <p><?php __e('Click <a href="http://devmx.de/en/teamspeak3-webviewer-berechtigungen-richtig-setzen" target="_blank">here</a> to see a tutorial about how to set the permissions for the guest account correctly. Set this option to no if you want to use the guest account.') ?>
                            </p></td>
                    </tr>
                    <tr id="config-username">
                        <td><?php __e('Username') ?></td>
                        <td><input type="text" value="<?php echo($data['username_value']) ?>" name="username" /></td>    
                        <td class="option-descr"><?php __e('The username of the query-user.') ?></td>
                    </tr>
                    <tr id="config-password">
                        <td><?php __e('Password') ?></td>
                        <td><input type="text" value="<?php echo($data['password_value']) ?>" name="password" /></td>
                        <td class="option-descr"><?php __e('Password corresponding to the username above.') ?></td>
                    </tr>
                </table>
            </div>

            <!-- Module -->
            <div id="tab2">
                <div class="ui-state-highlight ui-corner-all" style="margin-top: 20px; padding: 0 .7em; margin-bottom: 10px;"> 
                    <p><span class="ui-icon ui-icon-info" style="float: left; margin-right: .3em;"></span>
                        <?php __e('Click on a module to edit the modules configfile. Drag and sort the modules in the order you want them displayed') ?></p>
                </div>
                <div>
                    <table class="config" cellspacing="0">
                        <tr>
                            <td><div>
                                    <p><?php __e('enabled modules:') ?></p>
                                    <ul id="sort1" class="sortable">
                                        <?php foreach ($data['enabledModules'] as $mod) : ?>
                                            <li id="<?php echo $mod; ?>" class="module-active"><span class="module-edit" onclick="javascript: openModuleConfig('<?php echo $mod; ?>');"><?php echo $mod ?></span></li>
                                        <?php endforeach; ?>
                                    </ul>
                                </div></td>
                            <td><div>
                                    <p><?php __e('disabled modules:') ?></p>
                                    <ul id="sort2" class="sortable">
                                        <?php foreach ($data['disabledModules'] as $mod) : ?>
                                            <li id="<?php echo $mod; ?>" class="module-inactive"><span class="module-edit" onclick="javascript: openModuleConfig('<?php echo $mod ?>');"><?php echo $mod ?></span></li>
                                        <?php endforeach; ?>
                                    </ul>
                                </div></td>                      
                        </tr>
                        <tr>
                            <td><span class="button" onclick="javascript: enableAllModules();"><?php __e('Enable all'); ?></span></td>
                            <td><span class="button" onclick="javascript: disableAllModules();"><?php __e('Disable all'); ?></span></td>
                        </tr>
                    </table>
                </div>
                <input id="modules_hidden" type="hidden" name="module[]" value="" />
            </div>

            <!-- Style -->
            <div id="tab3">
                <table class="config style" cellspacing="0">
                    <tr id="servericons-config">
                        <td><?php __e('Download servericons automatically') ?></td>
                        <td>
                            <input id="servericons-true" type="radio" name="servericons" value="true" <?php if ($data['downloadIcons'] == "true" || $data['downloadIcons'] == '') : ?>checked="checked"><?php endif; ?><span> <?php __e('Enabled') ?></span><br>
                            <input id="servericons-false" type="radio" name="servericons" value="false" <?php if ($data['downloadIcons'] == "false") : ?>checked="checked"><?php endif; ?><span> <?php __e('Disabled') ?></span>
                        </td>
                        <td class="option-descr"><?php __e('If you set this on true, custom icons will be downloaded automatically.') ?></td>
                    </tr>
                    <tr id="imagepack-config">
                        <td><?php __e('Imagepack for icons') ?></td>
                        <td><?php echo($data['imagepack_html']) ?></td>
                        <td class="option-descr"><?php __e('If you disable the automatic icon download the imagepack selected here will be used.') ?></td>
                    </tr>
                    <tr id="stylesheet-config">
                        <td><?php __e('Stylesheet') ?></td>
                        <td><?php echo($data['style_html']) ?></td>
                        <td class="option-descr"><?php __e('The stylesheet which should be used for the viewer') ?></td>
                    </tr>
                    <tr id="arrow-config">
                        <td><?php __e('Display arrows') ?></td>
                        <td><?php echo($data['arrow_html']) ?></td>
                        <td class="option-descr"><?php __e('If you enable this, the viewer will show arrows next to the channel as in the client.') ?></td>
                    </tr>
                    <tr id="filter-config">
                        <td><?php __e('Display filter') ?></td>
                        <td><input type="radio" name="display-filter" value="standard" <?php if ($data['display-filter'] == "standard"): ?> checked="checked"<?php endif; ?>> <?php __e('Standard') ?><br>
                            <input type="radio" name="display-filter" value="clientsonly" <?php if ($data['display-filter'] == "clientsonly"): ?> checked="checked"<?php endif; ?>> <?php __e('Clientsonly') ?><br>
                            <input type="radio" name="display-filter" value="channelclientsonly" <?php if ($data['display-filter'] == "channelclientsonly"): ?> checked="checked" <?php endif; ?>> <?php __e('Channelclientsonly') ?></td>
                        <td class="option-descr">
                            <?php __e('standard: shows the viewer like in the TeamSpeak3 Client') ?><br>
                            <?php __e('clientsonly: shows only clients') ?><br>
                            <?php __e('channelclientsonly: shows only channels with clients inside') ?>
                        </td>
                    </tr>
                    <tr id="show-image-config">
                        <td><?php __e('Show images') ?></td>
                        <td>
                            <input type="radio" name="show_icons" value="true" <?php if ($data['show-images'] == "true") : ?> checked="checked"<?php endif; ?>> <?php __e('Yes') ?><br>
                            <input type="radio" name="show_icons" value="false" <?php if ($data['show-images'] == "false") : ?> checked="checked"<?php endif; ?>> <?php __e('No') ?><br>
                        </td>
                        <td class="option-descr"><?php __e('If you disable this, the webviewer will not show any images on the right side') ?></td>
                    </tr>
                    <tr id="show-country-icons-config">
                        <td><?php __e('Show country icons') ?></td>
                        <td>
                            <input type="radio" name="show_country_icons" value="true" <?php if ($data['show_country_icons'] == "true") : ?> checked="checked"<?php endif; ?>> <?php __e('Yes') ?><br>
                            <input type="radio" name="show_country_icons" value="false" <?php if ($data['show_country_icons'] == "false") : ?> checked="checked"<?php endif; ?>> <?php __e('No') ?>
                        </td>
                        <td class="option-descr"><?php __e('If country icons should be displayed as in the TeamSpeak3 Client') ?></td>
                    </tr>
                </table>
            </div>

            <!-- Caching -->
            <div id="tab4">
                <table class="config" cellspacing="0">
                    <tr>
                        <td><?php __e('Enable caching?') ?>
                        <td>
                            <?php echo($data['caching_html']) ?>
                        </td>
                        <td class="option-descr"><?php __e('If this on true the viewer will buffer the data it received. It is strongly recommended to enable caching due to stability reasons.') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Caching method') ?></td>
                        <td>
                            <input type="radio" name="caching-method" value="file" <?php if ($data['caching-method'] == "file" || $data['caching-method'] == "") : ?>checked="checked"<?php endif; ?> /> <?php __e('File Cache') ?><br>
                            <?php if (extension_loaded("apc")) : ?><input type="radio" name="caching-method" value="apc" <?php if ($data['caching-method'] == "apc") : ?>checked="checked" <?php endif; ?>/> <?php __e('APC-Cache') ?><?php endif; ?>
                        </td>
                        <td class="option-descr"><?php __e('The caching-method the viewer should use for caching. Some options may not be available on our webspace (Recommended: File-Cache).') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Standard cachetime') ?></td>
                        <td>
                            <?php echo($data['standard_caching_html']) ?>
                        </td>
                        <td class="option-descr"><?php __e('The time in seconds the viewer should buffer the serverdata (Recommended: 60).') ?></td>
                    </tr>
                </table>
            </div>

            <!-- Stuff -->
            <div id="tab5">
                <table class="config language" cellspacing="0">
                    <tr>
                        <td><?php __e('Language') ?></td>
                        <td><?php echo($data['language_html']) ?></td>
                        <td class="option-descr"><?php __e('Language of the viewer') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Date/ Time format') ?></td>
                        <td><input type="text" name="date-format" value="<?php echo((string) $data['date_format']) ?>" /></td>
                        <td class="option-descr"><?php __e('Format of <a href="http://php.net/manual/en/function.date.php" target="_blank">PHPs date() function</a>') ?></td>
                    </tr>
                    <tr>
                        <td><?php __e('Provide usage statistics') ?></td>
                        <td>
                            <input type="radio" name="usage-statistics" value="true" <?php if ($data['config']->usage_stats == "true"): ?>checked="checked" <?php endif; ?>> <?php __e('Yes') ?><br>
                            <input type="radio" name="usage-statistics" value="false" <?php if ($data['config']->usage_stats != "true") : ?>checked="checked" <?php endif; ?>> <?php __e('No') ?>
                        </td>
                        <td class="option-descr"><?php __e('If enabled the url to your webviewer will be submitted to devMX') ?></td>
                    </tr>
                </table>
            </div>
        </div>

        <!-- Submit-Button -->
        <br>
        <input type="submit" value="<?php __e('Save configfile') ?>" />
    </form>
    <iframe style="display:none" id="module-config" allowTransparency="true" frameborder="0" scrolling="0"></iframe>

</div>
<span id="last">&nbsp;</span>
